src/libOpenImageIO/imagebufalgo_xform.cpp

```diff
@@ -1144,4 +1144,261 @@ ImageBufAlgo::rotate(const ImageBuf& src, float angle, string_view filtername,
+            dst.error("ImageBufAlgo::st_warp : Output and ST buffers must have "
+                      "the same full width and height");
+            return false;
+        }
+    }
+
+    // Prep the dest spec using the channels from `src`, and the intersection of
+    // `roi` and the ROI from the ST buffer (since the ST warp is only defined
+    // for pixels in the ST buffer). We grab a copy of the input ROI before
+    // `IBAprep`, since we want to intersect it ourselves.
+    ROI inputROI(roi);
+    bool res
+        = ImageBufAlgo::IBAprep(roi, &dst, &src,
+                                ImageBufAlgo::IBAprep_NO_SUPPORT_VOLUME
+                                    | ImageBufAlgo::IBAprep_NO_COPY_ROI_FULL);
+    if (res) {
+        roi         = roi_intersection(inputROI, stSpec.roi());
+        roi.chbegin = std::max(roi.chbegin, 0);
+        roi.chend   = std::min(roi.chend, src.spec().nchannels);
+
+        ImageSpec destSpec(dst.spec());
+        destSpec.set_roi(roi);
+        destSpec.set_roi_full(stSpec.roi_full());
+        // XXX: It would be nice to be able to tell `IBAprep` to skip the buffer
+        // initialization. Worth adding a new prep flag?
+        dst.reset(destSpec);
+    }
+    return res;
+}
+
+
+
+bool
+ImageBufAlgo::st_warp(ImageBuf& dst, const ImageBuf& src, const ImageBuf& stbuf,
+                      Filter2D* filter, int chan_s, int chan_t, bool flip_s,
+                      bool flip_t, ROI roi, int nthreads)
+{
+    pvt::LoggedTimer logtime("IBA::st_warp");
+
+    if (!check_st_warp_args(dst, src, stbuf, chan_s, chan_t, roi)) {
+        return false;
+    }
+
+    bool ok;
+    OIIO_DISPATCH_COMMON_TYPES3(ok, "st_warp", st_warp_, dst.spec().format,
+                                src.spec().format, stbuf.spec().format, dst,
+                                src, stbuf, chan_s, chan_t, flip_s, flip_t,
+                                filter, roi, nthreads);
+    return ok;
+}
+
+
+
+bool
+ImageBufAlgo::st_warp(ImageBuf& dst, const ImageBuf& src, const ImageBuf& stbuf,
+                      string_view filtername, float filterwidth, int chan_s,
+                      int chan_t, bool flip_s, bool flip_t, ROI roi,
+                      int nthreads)
+{
+    pvt::LoggedTimer logtime("IBA::st_warp");
+
+    // XXX: As with other IBA functions that take a filter name and width, we
+    // will end up validating the arguments twice on this code path, but we want
+    // to know the scale factor between the source and ST/dest buffers so we can
+    // use `get_resize_filter`, and the overhead should be negligible.
+    if (!check_st_warp_args(dst, src, stbuf, chan_s, chan_t, roi)) {
+        return false;
+    }
+
+    // Resize ratios
+    float wratio = float(dst.spec().full_width) / src.spec().full_width;
+    float hratio = float(dst.spec().full_height) / src.spec().full_height;
+
+    auto filter = get_resize_filter(filtername, filterwidth, dst, wratio,
+                                    hratio);
```

Comment:
I might end up canning the use of `get_resize_filter` in favor of a uniform default filter, at least in part to remove the extra call to `check_st_warp_args`.

Reply:
OK, I ended up pulling out a static `get_warp_filter` helper, which is like `get_resize_filter`, but always defaults to `lanczos3`, and is used by both `warp` and `st_warp`.